### PR TITLE
fix export for enum and set types for mysql

### LINKED ISSUE #251: [BUG] MYSQL: incorrect syntax for SET and ENUM export SQL
When creating a MYSQL database creating a table with an `ENUM` or `SET` column type the exported SQL file (under file > export SQL) is incorrect.
Whether valid, invalid, or empty values are entered in the drawdb column's values input.
Currently, the exported syntax is

```mysql
CREATE TABLE .... (
  `enum_column` ENUM ...,
  `set_column` SET ...,
  ...
);
```

*missing brackets containing the enum/set's values*
- [enum column syntax](https://dev.mysql.com/doc/refman/8.4/en/enum.html#enum-using:~:text=size%20ENUM(%27x%2Dsmall%27%2C%20%27small%27%2C%20%27medium%27%2C%20%27large%27%2C%20%27x%2Dlarge%27))
- [set column syntax](https://dev.mysql.com/doc/refman/8.0/en/set.html#:~:text=mysql%3E%20CREATE%20TABLE%20myset%20(col%20SET(%27a%27%2C%20%27b%27%2C%20%27c%27%2C%20%27d%27))%3B)

the expected output is...

```mysql
CREATE TABLE .... (
  `enum_column` ENUM('value','val',...) ...,
  `set_column` SET('a','1',...) ...,
  ...
);
```

## To Reproduce
Steps to reproduce the behaviour:
1. Go to drawdb editor
2. Choose a MySQL database
3. Create a new table
4. Add a column of set or enum type
5. Add values for the column
6. Click on file > 'Export SQL'
7. See erroneous create table statement

## **Desktop:**
 - OS: Microsoft Windows 11 Home
 - Browser Microsoft Edge

### Additional context

Not sure if the MySQL export is for a different version of MySQL (I'm using MySQL V8.0.30), ~~I have a pr to solve the issue that I will link to this issue.~~

~~[I'm happy to open a PR with a fix to this issue](https://github.com/drawdb-io/drawdb/commit/624bd1ed2c7977b665fd85cd56869512cf22ce71)~~

### PR DESCRIPTION
**Pull request by csc530**
fixes #251

I tested it with both enum and set types and gives the expected output.

It also shouldn't interfere with any other column types when exporting: *as long as their `field` does not contain a `values` property.*

Please, let me know if any changes or tests need to be made

### CODE REVIEW

File: src/utils/exportSQL/mysql.js [hunk 1]
```diff
@@ -3,13 +3,14 @@ import { parseDefault } from "./shared";
 import { dbToTypes } from "../../data/datatypes";
 
 export function toMySQL(diagram) {
+  console.log(diagram)
```
Comment (1ilit) [inline src/utils/exportSQL/mysql.js:6]: Could you please remove this

File: src/utils/exportSQL/mysql.js [hunk 2]
```diff
@@ -3,13 +3,14 @@ import { parseDefault } from "./shared";
...
   return `${diagram.tables
     .map(
       (table) =>
         `CREATE TABLE \`${table.name}\` (\n${table.fields
           .map(
             (field) =>
-              `\t\`${field.name}\` ${field.type}${field.unsigned ? " UNSIGNED" : ""}${field.size !== undefined && field.size !== "" ? "(" + field.size + ")" : ""}${
+              `\t\`${field.name}\` ${field.type}${field.values ? "(" + field.values.map(value=>"'" + value + "'").join(", ")  + ")" : ""}${field.unsigned ? " UNSIGNED" : ""}${field.size !== undefined && field.size !== "" ? "(" + field.size + ")" : ""}${
```
Comment (1ilit) [inline src/utils/exportSQL/mysql.js:13]: Also, please format. If you're on vscode you can use alt+shift+f

**Review by 1ilit — CHANGES_REQUESTED** [review]
Thanks for the contribution. After you fix the comments could you also add this to mariadb as well?

File: src/utils/exportSQL/mysql.js (hunk 2 above)
Comment (csc530) [reply src/utils/exportSQL/mysql.js:13]: ok, I'm using the default "typescript and javascript language features" formatter in vscode

File: src/utils/exportSQL/mysql.js (hunk 1 above)
Comment (csc530) [reply src/utils/exportSQL/mysql.js:6]: oops😅

**Comment by Amirbakhtiaro** [comment]
> fixes #251
> 
> I tested it with both enum and set types and gives the expected output.
> 
> It also shouldn't interfere with any other column types when exporting: _as long as their `field` does not contain a `values` property._
> 
> Please, let me know if any changes or tests need to be made

https://github.com/drawdb-io/drawdb/pull/252#issue-2560506840

**Comment by Amirbakhtiaro** [comment]
Hi

**Pull request merged**